patroni/postgresql.py

```diff
@@ -69,6 +70,23 @@ def __init__(self, config):
         self._cursor_holder = None
         self.members = []  # list of already existing replication slots
         self.retry = Retry(max_tries=-1, deadline=10, max_delay=1, retry_exceptions=PostgresConnectionException)
+        self.init_pg_rewind()
+
+    def init_pg_rewind(self):
+        try:
+            self._pg_rewind_present = ('username' in self._pg_rewind and
```

Comment:
We should explore the possibility of checking if pg_rewind failed because this options are not set and distinguish it from other failures. Rationale: those options might not be set in yam config, but in postgresql.conf, and we cannot check it at this stage.